### ISSUE TITLE
Better Error Message  For Space Allocation Error

### ISSUE THREAD

**Issue by quasiben** [rapidsai/ucx-py#163]
As I test I loaded 24GB onto a GPU with 32GB max:

```python
In [1]: import cudf

In [2]: import numpy as np

In [3]: size = 1_000_000_000

In [4]: cdf = cudf.DataFrame({"a": np.random.random(size), "b": np.random.random(size)}, index=np.random.randint(size, size = size))
```

Then I tried to send-recv with ~4GB of data.  This was done by amending `test_send_recv_cudf` to use 2 ** 29 bytes in test 6: `(cudf.DataFrame({"a": range(2 ** 29)})`

The result was a segfault:

```
tests/test_custom_ep_send_recv.py::test_send_recv_cudf[<lambda>6] starting server...
Sending meta...
Receiving frames...
[1566509989.056431] [dgx15:81618:0]         ucp_mm.c:111  UCX  ERROR failed to register address (nil) length 4294967296 on md[2]=ib/mlx5_3: Invalid parameter
[1566509989.056449] [dgx15:81618:0]    ucp_request.c:264  UCX  ERROR failed to register user buffer datatype 0x8 address (nil) len 4294967296: Invalid parameter
[dgx15:81618:0:81618] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace ====
    0  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/libucs.so.0(+0x2215f) [0x7f2f1a54315f]
    1  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/libucs.so.0(+0x222bd) [0x7f2f1a5432bd]
    2  /lib/x86_64-linux-gnu/libc.so.6(+0x14e156) [0x7f2f1d582156]
    3  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/libucp.so.0(ucp_rndv_data_handler+0x2ec) [0x7f2f1ac0a552]
    4  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/ucx/libuct_ib.so.0(+0x604dd) [0x7f2ebd9f44dd]
    5  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/ucx/libuct_ib.so.0(uct_rc_mlx5_iface_progress+0x636) [0x7f2ebd9f5015]
    6  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/libucp.so.0(+0x27be7) [0x7f2f1abd3be7]
    7  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/libucp.so.0(ucp_worker_progress+0x137) [0x7f2f1abd91f6]
    8  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0xc70dc) [0x7f2f1af220dc]
    9  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0x6a195) [0x7f2f1aec5195]
   10  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0x2ab9d) [0x7f2f1ae85b9d]
   11  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0x4bda4) [0x7f2f1aea6da4]
   12  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(PyCFunction_Call+0x65) [0x55ff26864b65]
   13  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0x233e0) [0x7f2f1ae7e3e0]
   14  /home/nfs/bzaitlen/GitRepos/ucx-py/ucp/_libs/ucp_py.cpython-37m-x86_64-linux-gnu.so(+0x4b3cd) [0x7f2f1aea63cd]
   15  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyMethodDef_RawFastCallKeywords+0x305) [0x55ff268a1785]
   16  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyCFunction_FastCallKeywords+0x21) [0x55ff268a1801]
   17  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x544e) [0x55ff268fd87e]
   18  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyGen_Send+0x14c) [0x55ff2689a51c]
   19  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/lib/python3.7/lib-dynload/_asyncio.cpython-37m-x86_64-linux-gnu.so(+0xabc9) [0x7f2f1b19dbc9]
   20  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyObject_FastCallKeywords+0x49b) [0x55ff268a1d2b]
   21  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(+0x20abc3) [0x55ff26932bc3]
   22  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyMethodDef_RawFastCallDict+0x199) [0x55ff268648b9]
   23  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyCFunction_FastCallDict+0x21) [0x55ff26864ae1]
   24  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x5d21) [0x55ff268fe151]
   25  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0xfb) [0x55ff268a079b]
   26  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x6a0) [0x55ff268f8ad0]
   27  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0xfb) [0x55ff268a079b]
   28  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x6a0) [0x55ff268f8ad0]
   29  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0xfb) [0x55ff268a079b]
   30  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x6a0) [0x55ff268f8ad0]
   31  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0xfb) [0x55ff268a079b]
   32  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x6a0) [0x55ff268f8ad0]
   33  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalCodeWithName+0xbb9) [0x55ff2683edb9]
   34  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallDict+0x1d5) [0x55ff2683f5d5]
   35  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x1e20) [0x55ff268fa250]
   36  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalCodeWithName+0xbb9) [0x55ff2683edb9]
   37  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0x387) [0x55ff268a0a27]
   38  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x14ce) [0x55ff268f98fe]
   39  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalCodeWithName+0x2f9) [0x55ff2683e4f9]
   40  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0x387) [0x55ff268a0a27]
   41  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x4aa9) [0x55ff268fced9]
   42  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallKeywords+0xfb) [0x55ff268a079b]
   43  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalFrameDefault+0x4aa9) [0x55ff268fced9]
   44  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyEval_EvalCodeWithName+0x2f9) [0x55ff2683e4f9]
   45  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyFunction_FastCallDict+0x400) [0x55ff2683f800]
   46  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyObject_Call_Prepend+0x63) [0x55ff26856c43]
   47  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(+0x17116a) [0x55ff2689916a]
   48  /home/nfs/bzaitlen/miniconda3/envs/cudf-dev/bin/python(_PyObject_FastCallKeywords+0x49b) [0x55ff268a1d2b]

```

Note: Without any data loaded on a GPU this test passes.

Could ucx capture this segfault and message appropriate that space has run out ?

**Comment by quasiben** [comment]
Maybe this line is saying it can't allocated enough memory:
> ERROR failed to register address (nil) length 4294967296 on md[2]=ib/mlx5_3: Invalid parameter

**Comment by madsbk** [comment]
@quasiben, this should be fixed by #190. Can you check if you still get the segfault?

**Comment by quasiben** [comment]
No segfault!

> out of memory to allocate 4294967296 bytes (total 8589934592 bytes)
